### PR TITLE
fix: Added CI back for `check_no_std` with compilation to `wasm32`

### PR DESCRIPTION
**Pull request by puma314**
Previously the CI check for `check_no_std` was commented out, as the `riscv32imac-unknown-none-elf` target doesn't come with `std` and there are many dependencies in `reth` that don't support `no_std`, making it difficult to compile to this target.

However, there are several embedded environments (namely zkVMs like SP1) that *do* support `std`, but do not (easily) support crates with embedded C++ dependencies (like secp256k1, or blst or z-std) and also do not support networking crates like socket. The zkVM embedded environment is therefore similar to a target like `wasm32-wasi` which comes with `std`, but also does not support crates like secp256k1, blst and does not support networking crates. 

This PR adds back the `no_std` CI check but with the `wasm32-wasi` target, which should be significantly easier to support within Reth and also will ensure that all relevant packages that might be used in a zkVM context can compile. 

With this PR, `reth-codecs` and `reth-primitives-traits` compiles with `no-default-features` to `wasm32-wasi`. `reth-primitives` _almost_ compiles with `no-default-features`, but I'm running into this weird error that I was hoping @JackG-eth could help with since I saw he did the replacement of `thiserror_no_std::Error` everywhere.

```
Compiling reth-primitives v1.0.0 (/Users/umaroy/Documents/reth/crates/primitives)
error[E0433]: failed to resolve: use of undeclared crate or module `std`
 --> crates/primitives/src/transaction/error.rs:6:1
  |
6 | pub enum InvalidTransactionError {
  | ^^^ use of undeclared crate or module `std`
  |
help: consider importing this module
  |
1 + use core::error;

error[E0433]: failed to resolve: use of undeclared crate or module `std`
  --> crates/primitives/src/transaction/error.rs:59:1
   |
59 | pub enum TransactionConversionError {
   | ^^^ use of undeclared crate or module `std`
   |
help: consider importing this module
   |
1  + use core::error;
   |

error[E0433]: failed to resolve: use of undeclared crate or module `std`
  --> crates/primitives/src/transaction/error.rs:70:1
   |
70 | pub enum TryFromRecoveredTransactionError {
   | ^^^ use of undeclared crate or module `std`
   |
help: consider importing this module
   |
1  + use core::error;
```

### CODE REVIEW

File: crates/primitives/Cargo.toml [hunk 1]
```diff
@@ -33,7 +33,8 @@ secp256k1 = { workspace = true, features = [
     "global-context",
     "recovery",
     "rand",
-] }
+], optional = true }
+k256 = { version = "0.13.3", default-features = false, features = ["ecdsa"] }
```
Comment (mattsse) [inline crates/primitives/Cargo.toml:37]: why do we need this if we're not using it?

**Review by mattsse — CHANGES_REQUESTED** [review]
this currently does not compile without default features because we need to use this combo:

thiserror-no-std = { workspace = true, default-features = false }

std = ["thiserror-no-std/std"]

File: .github/assets/check_no_std.sh [hunk 2]
```diff
@@ -3,21 +3,21 @@ set -eo pipefail
...
 #   reth-evm-ethereum
 #   reth-network-peers
-#   reth-primitives
-#   reth-primitives-traits
 #   reth-revm
 )
 
 for package in "${no_std_packages[@]}"; do
-  cmd="cargo +stable build -p $package --target riscv32imac-unknown-none-elf --no-default-features"
+  cmd="cargo +stable build -p $package --target wasm32-wasi --no-default-features"
```
Comment (Rjected) [inline .github/assets/check_no_std.sh:20]:
```suggestion
  cmd="cargo +stable build -p $package --target wasm32-wasip1 --no-default-features"
```
`wasm32-wasi` is being renamed to `wasm32-wasip1` and will be removed
https://github.com/rust-lang/compiler-team/issues/607

**Review by Rjected — COMMENTED** [review]
This is cool! It would be great to have this in CI

**Comment by shekhirin** [comment]
Superseded by https://github.com/paradigmxyz/reth/pull/9430 and https://github.com/paradigmxyz/reth/pull/9982

**Pull request closed without merging**